assignment/tool/src/webapp/vm/assignment/chef_assignments_student_view_assignment_honor_pledge.vm

```diff
@@ -12,7 +12,14 @@
             #assignmentTitleIcon($assignment)
             $validator.escapeHtml($!assignment.Title)
         </h3>
-        #progressBar($!service.getProgressBarStatus($!submission))
+
+        ##special case, need to add honor and keep original order
+        #set($statusMap = $!service.getProgressBarStatus($!submission))
+        #set($statusMapAll = $statusMap.getClass().newInstance())
+        $!statusMapAll.put($!tlang.getString("gen.hpsta"), false)
+        $!statusMapAll.putAll($statusMap)
+        #progressBar($statusMapAll)
```

Comment:
statusMapAll should go in the action, imo. There's already too much logic in the templates.